### ISSUE TITLE
Bundling is obsolete and detrimental, CDN should be favored instead

### ISSUE THREAD

**Issue by Xeevis** [abpframework/abp#1493]
I like what bundling in ABP has to offer. I don't like how it's being used. Devil is in the 
concatenation.  It's obsolete workaround useful only to reduce HTTP requests, the most obvious shortcoming of legacy HTTP/1 protocol. For HTTP/2, which is now pretty much mainstream, this workaround is not just not helping anything, **it's making everything worse**:

Tiniest of changes to any CSS/JS asset means all users need to redownload whole big bundle again and again, tremendously increasing bandwidth usage and loading times. Since HTTP/2 downloads all assets in parallel, it's desired to have more smaller files where client can download only ones not in local cache. 

With that said, focus should be put on CDNs instead, where performance improvements can be made. CDN provides copies from servers nearest to the client and client browser can use these cached libraries across multiple websites, reducing number of assets it has to download on first visit.

I can disable concatenation (which I believe should be default behavior) with

```c#
Configure<BundlingOptions>(options =>
{
    options.Mode = BundlingMode.None;
});
```

But I can't figure out how to implement CDNs with ABP bundles, had to sacrifice modularity of bundles for performance of CDNs with default ASP.NET Core's tag helpers.

```html
<environment include="Development">
    <script src="~/lib/jquery/jquery.js"></script>
    <script src="~/lib/popper.js/umd/popper.js"></script>
    <script src="~/lib/bootstrap/js/bootstrap.js"></script>           
</environment>
<environment exclude="Development">
    <script src="https://cdn.jsdelivr.net/npm/jquery@3.4.1/dist/jquery.min.js" 
            integrity="sha384-vk5WoKIaW/vJyUAd9n/wmopsmNhiy+L2Z+SBxGYnUkunIxVxAv/UtMOhba/xskxh" crossorigin="anonymous"
            asp-fallback-src="~/lib/jquery/jquery.min.js"
            asp-fallback-test="window.jQuery"
            asp-suppress-fallback-integrity="true"></script>
    <script src="https://cdn.jsdelivr.net/npm/popper.js@1.15.0/dist/umd/popper.min.js"
            integrity="sha384-L2pyEeut/H3mtgCBaUNw7KWzp5n9+4pDQiExs933/5QfaTh8YStYFFkOzSoXjlTb" crossorigin="anonymous"
            asp-fallback-src="~/lib/popper.js/umd/popper.min.js"
            asp-fallback-test="window.Popper"
            asp-suppress-fallback-integrity="true"></script>
    <script src="https://cdn.jsdelivr.net/npm/bootstrap@4.3.1/dist/js/bootstrap.min.js"
            integrity="sha384-JjSmVgyd0p3pXB1rRibZUAYoIIy6OrQ6VrjIEaFf/nJGzIxFDsf4x0xIM+B07jRM" crossorigin="anonymous"
            asp-fallback-src="~/lib/bootstrap/js/bootstrap.min.js"
            asp-fallback-test="window.jQuery && window.jQuery.fn && window.jQuery.fn.modal"
            asp-suppress-fallback-integrity="true"></script>
</environment>
```

**Comment by hikalkan** [comment]
>I like what bundling in ABP has to offer

Thanks :)

>focus should be put on CDNs instead, where performance improvements can be made. CDN provides copies from servers nearest to the client and client browser can use these cached libraries across multiple websites, reducing number of assets it has to download on first visit.

That's good, OK. But most of the applications are not using CDNs. I don't thing the bundling should be disabled by default.

We can not remove bundling system since it is essential for modularity. You probably know it.

CDN support is important and it is already covered in https://github.com/abpframework/abp/issues/294

I am closing this as duplicate. Follow the related issue. Thanks.
